ui/src/shared/components/data-loader-dropdown.tsx

```diff
@@ -1,24 +1,86 @@
 import {DataLoader} from 'argo-ui/src/components/data-loader';
 import {Select, SelectOption} from 'argo-ui/src/components/select/select';
 import * as React from 'react';
-import {useState} from 'react';
+import {useEffect, useRef, useState} from 'react';
 
 export function DataLoaderDropdown(props: {load: () => Promise<(string | SelectOption)[]>; onChange: (value: string) => void; placeholder?: string}) {
     const [selected, setSelected] = useState('');
+    const selectRef = useRef<HTMLDivElement>(null);
+    const [options, setOptions] = useState<(string | SelectOption)[]>([]);
+
+    const enhanceOptionsWithTitles = (options: (string | SelectOption)[]) => {
+        setOptions(options);
+        return options.map(option => {
+            if (typeof option === 'string') {
+                return {
+                    value: option,
+                    title: option
+                };
+            } else {
+                return option;
+            }
+        });
+    };
+
+    useEffect(() => {
+        const addTitleAttributes = () => {
+            setTimeout(() => {
+                const optionElements = document.querySelectorAll('.select__option');
+                optionElements.forEach(element => {
+                    const text = element.textContent || '';
+                    if (text) {
+                        element.setAttribute('title', text);
+                    }
+                });
+            }, 100);
+        };
+
+        if (options.length > 0) {
+            addTitleAttributes();
+        }
+    }, [options]);
+
+    useEffect(() => {
+        const handleDropdownOpen = () => {
+            const observer = new MutationObserver(mutations => {
+                mutations.forEach(mutation => {
+                    if (mutation.type === 'childList' && mutation.addedNodes.length > 0) {
+                        const optionElements = document.querySelectorAll('.select__option');
+                        optionElements.forEach(element => {
+                            const text = element.textContent || '';
+                            if (text) {
+                                element.setAttribute('title', text);
```

Comment:
Modifying DOM elements outside React is liable to cause bugs ([source](https://react.dev/learn/manipulating-the-dom-with-refs#best-practices-for-dom-manipulation-with-refs)):
>  Avoid changing DOM nodes managed by React. Modifying, adding children to, or removing children from elements that are managed by React can lead to inconsistent visual results or crashes like above.

Also, I don't see understand this is necessary. Why can't the `title` attribute be set at the time the `<Select>` component is instantiated?

Reply:
Thank you for your review.
I was not aware of that rule 😢 
I will make the necessary corrections, so please wait a moment.